Ingress/controllers/nginx-third-party/nginx/command.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nginx
+
+import (
+	"os"
+	"os/exec"
+
+	"github.com/golang/glog"
+)
+
+// Start starts a nginx (master process) and waits. If the process ends
+// we need to kill the current controller process and return the reason.
+func (ngx *NginxManager) Start() {
+	cmd := exec.Command("nginx")
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	if err := cmd.Start(); err != nil {
+		glog.Errorf("nginx error: %v", err)
+	}
+
+	if err := cmd.Wait(); err != nil {
+		glog.Fatalf("nginx error: %v", err)
```

Comment:
dont fatal (here and below)? also makes it really hard to unittest failure modes. Send events instaed, we can mock that out.